WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/ShippingLabelFormViewModel.swift

```diff
@@ -26,6 +26,37 @@ final class ShippingLabelFormViewModel {
     private(set) var destinationAddress: ShippingLabelAddress?
     private(set) var packagesResponse: ShippingLabelPackagesResponse?
     private(set) var selectedPackageID: String?
+    var selectedPackage: ShippingLabelPackageSelected? {
+        guard let packagesResponse = packagesResponse else {
+            return nil
+        }
+
+        for customPackage in packagesResponse.customPackages {
+            if customPackage.title == selectedPackageID {
+                return ShippingLabelPackageSelected(boxID: customPackage.title,
+                                                    length: customPackage.getLength(),
+                                                    width: customPackage.getWidth(),
+                                                    height: customPackage.getHeight(),
+                                                    weight: customPackage.getWidth(),
+                                                    isLetter: customPackage.isLetter)
+            }
+        }
+
+        for option in packagesResponse.predefinedOptions {
+            for predefinedPackage in option.predefinedPackages {
+                if predefinedPackage.id == selectedPackageID {
+                    return ShippingLabelPackageSelected(boxID: predefinedPackage.id,
+                                                        length: predefinedPackage.getLength(),
+                                                        width: predefinedPackage.getWidth(),
+                                                        height: predefinedPackage.getHeight(),
+                                                        weight: predefinedPackage.getWidth(),
+                                                        isLetter: predefinedPackage.isLetter)
+                }
+            }
+        }
```

Comment:
Not quite sure how to make this more expressive. It's tricky. 😄 

This is what I got so far but I haven't tested it! 

```swift
        // The lazy is critical for performance
        if let predefinedPackage = packagesResponse.predefinedOptions.lazy.compactMap({ option in
            option.predefinedPackages.first(where: { $0.id == self.selectedPackageID })
        }).first {
            return ShippingLabelPackageSelected(boxID: predefinedPackage.id,
                                                length: predefinedPackage.getLength(),
                                                width: predefinedPackage.getWidth(),
                                                height: predefinedPackage.getHeight(),
                                                weight: predefinedPackage.getWidth(),
                                                isLetter: predefinedPackage.isLetter)
        }
```

I'm not convinced this is better. 🤣 And I don't think it even works.

Reply:
I tried this version, a "combo" between my version and yours 😄 

```
        for option in packagesResponse.predefinedOptions {
            if let predefinedPackage = option.predefinedPackages.first(where: { $0.id == selectedPackageID }) {
                    return ShippingLabelPackageSelected(boxID: predefinedPackage.id,
                                                        length: predefinedPackage.getLength(),
                                                        width: predefinedPackage.getWidth(),
                                                        height: predefinedPackage.getHeight(),
                                                        weight: predefinedPackage.getWidth(),
                                                        isLetter: predefinedPackage.isLetter)
            }
        }
```